cppapi/server/attribute.cpp

```diff
@@ -4024,12 +4029,16 @@ void Attribute::fire_change_event(DevFailed *except)
 			{
 				force_change = true;
 			}
+			}
 
 			std::vector<std::string> filterable_names;
 			std::vector<double> filterable_data;
 			std::vector<std::string> filterable_names_lg;
 			std::vector<long> filterable_data_lg;
 
+			{
+			omni_mutex_lock oml(EventSupplier::get_event_mutex());
+
 			if (except != NULL)
 			{
 				prev_change_event.err    = true;
```

Comment:
* protecting access to prev_change_event, prev_archive_event and prev_quality_event
* again, we are taking the lock from another class to protect own data, it this ok?
* maybe we should add `Attribute::set_prev_change_event(const LastAttrValue&)` which will lock some mutex? Which mutex?

Race between readthread, here in this scope (line 4054 in this PR, no mutex) and polling thread (eventsupplier.cpp:416) - polling thread holds attr_mutex and event_mutex:
https://github.com/tango-controls/cppTango/blob/6a2857c32baba5454d936815279314e60b838b9b/cppapi/server/attribute.cpp#L4043-L4046
https://github.com/tango-controls/cppTango/blob/6a2857c32baba5454d936815279314e60b838b9b/cppapi/server/eventsupplier.cpp#L412-L414

```
==================
WARNING: ThreadSanitizer: data race (pid=3222)
  Write of size 8 at 0x7b040000a1a0 by thread T8:
    #0 operator delete[](void*) <null> (libtsan.so.0+0x742ad)
    #1 _CORBA_Sequence<double>::freebuf(double*) /usr/include/omniORB4/seqTemplatedecls.h:89 (libtango.so.9+0x5258fd)
    #2 _CORBA_Sequence<double>::~_CORBA_Sequence() /usr/include/omniORB4/seqTemplatedecls.h:127 (libtango.so.9+0x5281d6)
    #3 _CORBA_Unbounded_Sequence<double>::~_CORBA_Unbounded_Sequence() /usr/include/omniORB4/seqTemplatedecls.h:253 (libtango.so.9+0x524c2b)
    #4 _CORBA_Unbounded_Sequence_w_FixSizeElement<double, 8, 8>::~_CORBA_Unbounded_Sequence_w_FixSizeElement() /usr/include/omniORB4/seqTemplatedecls.h:516 (libtango.so.9+0x52196b)
    #5 Tango::DevVarDoubleArray::~DevVarDoubleArray() /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:233 (libtango.so.9+0x51b625)
    #6 Tango::AttrValUnion::_release_member() /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:3281 (libtango.so.9+0x592cf5)
    #7 Tango::AttrValUnion::double_att_value(Tango::DevVarDoubleArray const&) /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:3036 (libtango.so.9+0x592100)
    #8 Tango::AttrValUnion::operator=(Tango::AttrValUnion const&) /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:2860 (libtango.so.9+0x591947)
    #9 Tango::Attribute::fire_change_event(Tango::DevFailed*) /home/michal/Documents/cppTango/cppapi/server/attribute.cpp:4054 (libtango.so.9+0x79f312)
    #10 Tango::DeviceImpl::push_change_event(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double*, timeval&, Tango::AttrQuality, long, long, bool) /home/michal/Documents/cppTango/cppapi/server/dev_event.cpp:1120 (libtango.so.9+0x7d903c)
    #11 TestPushCrash_ns::readthread::run(void*) ../src/readthread.cpp:147 (TestPushCrash+0x34403)
    #12 omni_thread_wrapper /home/michal/Documents/omniORB-4.2.3/src/lib/omnithread/posix.cc:459 (libomnithread.so.4+0x3f34)
    #13 <null> <null> (libtsan.so.0+0x296ad)

  Previous write of size 8 at 0x7b040000a1a0 by thread T9 (mutexes: write M1262, write M1004):
    #0 operator new[](unsigned long) <null> (libtsan.so.0+0x73fda)
    #1 _CORBA_Sequence<double>::allocbuf(unsigned int) /usr/include/omniORB4/seqTemplatedecls.h:84 (libtango.so.9+0x528964)
    #2 _CORBA_Sequence<double>::copybuffer(unsigned int) /usr/include/omniORB4/seqTemplatedecls.h:205 (libtango.so.9+0x525767)
    #3 _CORBA_Sequence<double>::length(unsigned int) /usr/include/omniORB4/seqTemplatedecls.h:59 (libtango.so.9+0x52298a)
    #4 _CORBA_Sequence<double>::_CORBA_Sequence(_CORBA_Sequence<double> const&) /usr/include/omniORB4/seqTemplatedecls.h:147 (libtango.so.9+0x5ac977)
    #5 _CORBA_Unbounded_Sequence<double>::_CORBA_Unbounded_Sequence(_CORBA_Unbounded_Sequence<double> const&) /usr/include/omniORB4/seqTemplatedecls.h:242 (libtango.so.9+0x5a4867)
    #6 _CORBA_Unbounded_Sequence_w_FixSizeElement<double, 8, 8>::_CORBA_Unbounded_Sequence_w_FixSizeElement(_CORBA_Unbounded_Sequence_w_FixSizeElement<double, 8, 8> const&) /usr/include/omniORB4/seqTemplatedecls.h:503 (libtango.so.9+0x599ceb)
    #7 Tango::DevVarDoubleArray::DevVarDoubleArray(Tango::DevVarDoubleArray const&) /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:238 (libtango.so.9+0x590c43)
    #8 Tango::AttrValUnion::double_att_value(Tango::DevVarDoubleArray const&) /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:3040 (libtango.so.9+0x592162)
    #9 Tango::AttrValUnion::operator=(Tango::AttrValUnion const&) /home/michal/Documents/cppTango/cppapi/server/idl/tango.h:2860 (libtango.so.9+0x591947)
    #10 Tango::EventSupplier::detect_and_push_change_event(Tango::DeviceImpl*, Tango::EventSupplier::SuppliedEventData&, Tango::Attribute&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, Tango::DevFailed*, bool) /home/michal/Documents/cp
pTango/cppapi/server/eventsupplier.cpp:416 (libtango.so.9+0x89dfa4)
```